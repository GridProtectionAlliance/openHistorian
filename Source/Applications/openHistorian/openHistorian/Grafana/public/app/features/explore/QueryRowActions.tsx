import React from 'react';
import { Icon } from '@grafana/ui';

function formatLatency(value: number) {
  return `${(value / 1000).toFixed(1)}s`;
}

export type Props = {
  canToggleEditorModes: boolean;
  isDisabled?: boolean;
  isNotStarted: boolean;
  latency: number;
  onClickToggleEditorMode: () => void;
  onClickToggleDisabled: () => void;
  onClickRemoveButton: () => void;
};

export function QueryRowActions(props: Props) {
  const {
    canToggleEditorModes,
    onClickToggleEditorMode,
    onClickToggleDisabled,
    onClickRemoveButton,
    isDisabled,
    isNotStarted,
    latency,
  } = props;

  return (
    <div className="gf-form-inline flex-shrink-0">
      {canToggleEditorModes && (
        <div className="gf-form">
          <button
            aria-label="Edit mode button"
            className="gf-form-label gf-form-label--btn"
            onClick={onClickToggleEditorMode}
          >
            <Icon name="pen" />
          </button>
        </div>
      )}
      <div className="gf-form">
        <button disabled className="gf-form-label" title="Query row latency">
          {formatLatency(latency)}
        </button>
      </div>
      <div className="gf-form">
        <button
          disabled={isNotStarted}
          className="gf-form-label gf-form-label--btn"
          onClick={onClickToggleDisabled}
          title={isDisabled ? 'Enable query' : 'Disable query'}
        >
          <Icon name={isDisabled ? 'eye-slash' : 'eye'} />
        </button>
      </div>
      <div className="gf-form">
        <button className="gf-form-label gf-form-label--btn" onClick={onClickRemoveButton} title="Remove query">
          <Icon name="minus" />
        </button>
      </div>
    </div>
  );
}
