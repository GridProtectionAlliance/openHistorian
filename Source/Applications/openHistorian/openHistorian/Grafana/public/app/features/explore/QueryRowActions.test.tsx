import React from 'react';
import { QueryRowActions, Props } from './QueryRowActions';
import { shallow } from 'enzyme';

const setup = (propOverrides?: object) => {
  const props: Props = {
    isDisabled: false,
    isNotStarted: true,
    canToggleEditorModes: true,
    onClickToggleEditorMode: () => {},
    onClickToggleDisabled: () => {},
    onClickRemoveButton: () => {},
    latency: 0,
  };

  Object.assign(props, propOverrides);

  const wrapper = shallow(<QueryRowActions {...props} />);
  return wrapper;
};

describe('QueryRowActions', () => {
  it('should render component', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render component without editor mode', () => {
    const wrapper = setup({ canToggleEditorModes: false });
    expect(wrapper.find({ 'aria-label': 'Edit mode button' })).toHaveLength(0);
  });
  it('should change icon to eye-slash when query row result is hidden', () => {
    const wrapper = setup({ isDisabled: true });
    expect(wrapper.find({ title: 'Enable query' })).toHaveLength(1);
  });
  it('should change icon to eye when query row result is not hidden', () => {
    const wrapper = setup({ isDisabled: false });
    expect(wrapper.find({ title: 'Disable query' })).toHaveLength(1);
  });
});
