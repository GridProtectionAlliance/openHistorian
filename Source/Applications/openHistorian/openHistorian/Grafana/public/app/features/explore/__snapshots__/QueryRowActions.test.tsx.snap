// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryRowActions should render component 1`] = `
<div
  className="gf-form-inline flex-shrink-0"
>
  <div
    className="gf-form"
  >
    <button
      aria-label="Edit mode button"
      className="gf-form-label gf-form-label--btn"
      onClick={[Function]}
    >
      <Icon
        name="pen"
      />
    </button>
  </div>
  <div
    className="gf-form"
  >
    <button
      className="gf-form-label"
      disabled={true}
      title="Query row latency"
    >
      0.0s
    </button>
  </div>
  <div
    className="gf-form"
  >
    <button
      className="gf-form-label gf-form-label--btn"
      disabled={true}
      onClick={[Function]}
      title="Disable query"
    >
      <Icon
        name="eye"
      />
    </button>
  </div>
  <div
    className="gf-form"
  >
    <button
      className="gf-form-label gf-form-label--btn"
      onClick={[Function]}
      title="Remove query"
    >
      <Icon
        name="minus"
      />
    </button>
  </div>
</div>
`;
